### PR TITLE
Document params of `Application.make_handler()`.

### PR DESCRIPTION
**Pull request by f0t0n**
## What do these changes do?

Explicitly document parameters of `aiohttp.web.Application.make_handler()` method.

## Are there changes in behavior for the user?

Documentation became more clear.

## Related issue number

#1129

Also see this thread: https://github.com/KeepSafe/aiohttp/pull/1123#discussion_r76446415

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

### CODE REVIEW

**Comment by f0t0n** [comment]
@asvetlov Please check descriptions of parameters. I didn't find proper descriptions for some of them. Maybe you could point how to describe better. Also maybe we should drop some from the list.
At the moment on the built page they look like this:

image

File: docs/web_reference.rst [hunk 1]
```diff
@@ -1130,38 +1130,61 @@ duplicated like one using :meth:`Application.copy`.
...
-
-      ``secure_proxy_ssl_header`` keyword parameter
-      can be used to detect request scheme::
-
-         await loop.create_server(app.make_handler(
-            secure_proxy_ssl_header='X-Forwarded-Proto'),
-            '0.0.0.0', 8080)
+    Creates HTTP protocol factory for handling requests.
+
+    :param handler: Request handler. Default:
```
Comment (asvetlov) [inline docs/web_reference.rst]: Let's drop it from doc.
It's a hidden param, not part of public API (at least now now).

File: docs/web_reference.rst [hunk 2]
```diff
@@ -1130,38 +1130,61 @@ duplicated like one using :meth:`Application.copy`.
...
+        is deprecated in favor of :attr:`Application.debug`.
+        The :class:`Application`'s debug mode setting should be used
+        as a single point to setup a debug mode.
+
+    :param int max_line_size: Optional maximum header line size. Default:
+      ``8190``.
+    :param int max_headers: Optional maximum header size. Default: ``32768``.
+    :param int max_field_size: Optional maximum header field size. Default:
+      ``8190``.
+    :param eof_exc_class: Parameter of :class:`aiohttp.parsers.StreamParser`.
```
Comment (asvetlov) [inline docs/web_reference.rst]: Please drop it

File: docs/web_reference.rst [hunk 3]
```diff
@@ -1130,38 +1130,61 @@ duplicated like one using :meth:`Application.copy`.
...
+
+       app = Application(loop=loop)
+
+       # setup route table
+       # app.router.add_route(...)
+
+       await loop.create_server(app.make_handler(),
+                                '0.0.0.0', 8080)
+
+    ``secure_proxy_ssl_header`` keyword parameter
```
Comment (asvetlov) [inline docs/web_reference.rst]: Move it into `:param secure_proxy_ssl_header:` section.
Drop the code example, it's just wrong.
The value should be a tuple `secure_proxy_ssl_header=('X-Forwarded-Proto', 'https')` but I'm considering to deprecate it but support `X-Forwarded-Proto` implicitly in `web.Request.scheme`.

See also #1134

**Comment by f0t0n** [comment]
All the changes according to the comments in code review are done.

**Comment by asvetlov** [comment]
Thanks

**Pull request merged**